File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnVideoCredit.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* \class GnVideoCredit
*/
public class GnVideoCredit : GnDataObject {
  private HandleRef swigCPtr;

  internal GnVideoCredit(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnVideoCredit_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnVideoCredit obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnVideoCredit() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnVideoCredit(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

/**
*  Genre, e.g., comedy. This is a list/locale dependent, multi-level field.
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that a list or locale be loaded into memory.
*
*  To render locale-dependent information for list-based values, your application must allocate a
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information
*  is not set prior to a request for a list-based value.
*
* @param level :enum value specifying level value
* @return gnsdk_cstr_t
* @ingroup GDO_ValueKeys_Misc
*/
  public string Genre(GnDataLevel level) {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Genre(swigCPtr, (int)level);
    return ret;
  }

/**
*  Artist type. This is a list/locale dependent, multi-level field
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that a list or locale be loaded into memory.
*
*  To render locale-dependent information for list-based values, your application must allocate a
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information
*  is not set prior to a request for a list-based value.
*
* @param level        :enum value specifying level value
* @return gnsdk_cstr_t
* @ingroup GDO_ValueKeys_Misc
*/
  public string ArtistType(GnDataLevel level) {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_ArtistType(swigCPtr, (int)level);
    return ret;
  }

/**
*  Geographic location, e.g., "New York City". This is a list/locale dependent, multi-level field.
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that a list or locale be loaded into memory.
*
*  To render locale-dependent information for list-based values, your application must allocate a
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information
*  is not set prior to a request for a list-based value.
*
* @param level		[in] enum value specifying level value
* @return gnsdk_cstr_t
* @ingroup GDO_ValueKeys_Misc
*/
  public string Origin(GnDataLevel level) {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Origin(swigCPtr, (int)level);
    return ret;
  }

/**
* Artist era. This is a list/locale dependent, multi-level field.
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that a list or locale be loaded into memory.
*
*  To render locale-dependent information for list-based values, your application must allocate a
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information
*  is not set prior to a request for a list-based value.
*
* @param level		[in] enum value specifying level value
* @return gnsdk_cstr_t
* @ingroup GDO_ValueKeys_Misc
*/
  public string Era(GnDataLevel level) {
    string ret = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Era(swigCPtr, (int)level);
    return ret;
  }

/**
*  Role, e.g., Actor.
* <p><b>Note:</b></p>
*  For music credits, the absence of a role for a person indicates that person is the primary
*   artist, who may have performed multiple roles.
*  <p><b>Remarks:</b></p>
*  This is a list-based value requiring that a list or locale be loaded into memory.
*
*  To render locale-dependent information for list-based values, your application must allocate a
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information
*  is not set prior to a request for a list-based value.
* @ingroup GDO_ValueKeys_Role
*/
  public string Role {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Role_get(swigCPtr) );
	} 

  }

/**
* Role ID
* @ingroup GDO_ValueKeys_Role
*/
  public uint RoleId {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_RoleId_get(swigCPtr);
      return ret;
    } 
  }

/**
*  A number identifying the role's listing in the credits.
* @ingroup GDO_ValueKeys_Role
*/
  public string RoleBilling {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoCredit_RoleBilling_get(swigCPtr) );
	} 

  }

/**
*  The character's name on the show.
* @ingroup GDO_ValueKeys_Video
*/
  public string CharacterName {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnVideoCredit_CharacterName_get(swigCPtr) );
	} 

  }

/**
*  Role's rank in importance.
* @ingroup GDO_ValueKeys_Misc
*/
  public uint Rank {
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Rank_get(swigCPtr);
      return ret;
    } 
  }

/**
*  Official name object .
* @ingroup GDO_ChildKeys_Name
*/
  public GnName OfficialName {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_OfficialName_get(swigCPtr);
      GnName ret = (cPtr == IntPtr.Zero) ? null : new GnName(cPtr, true);
      return ret;
    } 
  }

/**
* Contributor object.
* @ingroup GDO_ChildKeys_Contributor
*/
  public GnContributor Contributor {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Contributor_get(swigCPtr);
      GnContributor ret = (cPtr == IntPtr.Zero) ? null : new GnContributor(cPtr, true);
      return ret;
    } 
  }

  public GnVideoSeasonEnumerable Seasons {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Seasons_get(swigCPtr);
      GnVideoSeasonEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoSeasonEnumerable(cPtr, true);
      return ret;
    } 
  }

  public GnVideoSeriesEnumerable Series {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Series_get(swigCPtr);
      GnVideoSeriesEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoSeriesEnumerable(cPtr, true);
      return ret;
    } 
  }

  public GnVideoWorkEnumerable Works {
    get {
      IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnVideoCredit_Works_get(swigCPtr);
      GnVideoWorkEnumerable ret = (cPtr == IntPtr.Zero) ? null : new GnVideoWorkEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnVideoCredit.java

package com.gracenote.gnsdk;

/** 
**/ 
 
public class GnVideoCredit extends GnDataObject {
  private long swigCPtr;

  protected GnVideoCredit(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnVideoCredit_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnVideoCredit obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnVideoCredit(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

/** 
*  Role, e.g., Actor. 
* <p><b>Note:</b></p> 
*  For music credits, the absence of a role for a person indicates that person is the primary 
*   artist, who may have performed multiple roles. 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that a list or locale be loaded into memory. 
* <p> 
*  To render locale-dependent information for list-based values, your application must allocate a 
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information 
*  is not set prior to a request for a list-based value. 
* 
*/ 
 
  public String role() {
    return gnsdk_javaJNI.GnVideoCredit_role(swigCPtr, this);
  }

/** 
* Role ID 
* 
*/ 
 
  public long roleId() {
    return gnsdk_javaJNI.GnVideoCredit_roleId(swigCPtr, this);
  }

/** 
*  A number identifying the role's listing in the credits. 
* 
*/ 
 
  public String roleBilling() {
    return gnsdk_javaJNI.GnVideoCredit_roleBilling(swigCPtr, this);
  }

/** 
*  The character's name on the show. 
* 
*/ 
 
  public String characterName() {
    return gnsdk_javaJNI.GnVideoCredit_characterName(swigCPtr, this);
  }

/** 
*  Role's rank in importance. 
* 
*/ 
 
  public long rank() {
    return gnsdk_javaJNI.GnVideoCredit_rank(swigCPtr, this);
  }

/** 
*  Genre, e.g., comedy. This is a list/locale dependent, multi-level field. 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that a list or locale be loaded into memory. 
* <p> 
*  To render locale-dependent information for list-based values, your application must allocate a 
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information 
*  is not set prior to a request for a list-based value. 
* <p> 
* @param level :enum value specifying level value 
* @return gnsdk_cstr_t 
* 
*/ 
 
  public String genre(GnDataLevel level) {
    return gnsdk_javaJNI.GnVideoCredit_genre(swigCPtr, this, level.swigValue());
  }

/** 
*  Artist type. This is a list/locale dependent, multi-level field 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that a list or locale be loaded into memory. 
* <p> 
*  To render locale-dependent information for list-based values, your application must allocate a 
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information 
*  is not set prior to a request for a list-based value. 
* <p> 
* @param level        :enum value specifying level value 
* @return gnsdk_cstr_t 
* 
*/ 
 
  public String artistType(GnDataLevel level) {
    return gnsdk_javaJNI.GnVideoCredit_artistType(swigCPtr, this, level.swigValue());
  }

/** 
*  Geographic location, e.g., "New York City". This is a list/locale dependent, multi-level field. 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that a list or locale be loaded into memory. 
* <p> 
*  To render locale-dependent information for list-based values, your application must allocate a 
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information 
*  is not set prior to a request for a list-based value. 
* <p> 
* @param level		[in] enum value specifying level value 
* @return gnsdk_cstr_t 
* 
*/ 
 
  public String origin(GnDataLevel level) {
    return gnsdk_javaJNI.GnVideoCredit_origin(swigCPtr, this, level.swigValue());
  }

/** 
* Artist era. This is a list/locale dependent, multi-level field. 
*  <p><b>Remarks:</b></p> 
*  This is a list-based value requiring that a list or locale be loaded into memory. 
* <p> 
*  To render locale-dependent information for list-based values, your application must allocate a 
*  <code>GnLocale</code> object. A <code>LocaleNotLoaded</code> message is returned when locale information 
*  is not set prior to a request for a list-based value. 
* <p> 
* @param level		[in] enum value specifying level value 
* @return gnsdk_cstr_t 
* 
*/ 
 
  public String era(GnDataLevel level) {
    return gnsdk_javaJNI.GnVideoCredit_era(swigCPtr, this, level.swigValue());
  }

/** 
*  Official name object . 
* 
*/ 
 
  public GnName officialName() {
    return new GnName(gnsdk_javaJNI.GnVideoCredit_officialName(swigCPtr, this), true);
  }

/** 
* Contributor object. 
* 
*/ 
 
  public GnContributor contributor() {
    return new GnContributor(gnsdk_javaJNI.GnVideoCredit_contributor(swigCPtr, this), true);
  }

  public GnVideoWorkIterable works() {
    return new GnVideoWorkIterable(gnsdk_javaJNI.GnVideoCredit_works(swigCPtr, this), true);
  }

  public GnVideoSeriesIterable series() {
    return new GnVideoSeriesIterable(gnsdk_javaJNI.GnVideoCredit_series(swigCPtr, this), true);
  }

  public GnVideoSeasonIterable seasons() {
    return new GnVideoSeasonIterable(gnsdk_javaJNI.GnVideoCredit_seasons(swigCPtr, this), true);
  }

}
